src/react/databrowser/objects/__tests__/Tags.test.jsx

```diff
@@ -0,0 +1,124 @@
+        component.find('input#tags-input-value').simulate('change', {
+            target: {
+                name: 'value',
+                value: 'value1',
+            },
+        });
+
+        // click on add button
+        component.find(AddButton).simulate('click');
+
+        // check if new item row added
+        expect(component.find(Item)).toHaveLength(2);
+
+        // check if function when pressing save button is triggered
+        expect(putObjectTaggingMock).toHaveBeenCalledTimes(0);
+        component.find('Button#tags-button-save').simulate('click');
+        expect(putObjectTaggingMock).toHaveBeenCalledTimes(1);
+    });
+
+    it('remove button and add button should be disabled', () => {
+        const { component } = reduxMount(<Tags objectMetadata={OBJECT_METADATA}/>);
+
+        // check if only one item row is rendered
+        const items = component.find(Item);
+        expect(items).toHaveLength(1);
+
+        // check if item row values are rendered
+        const firstItem = items.first();
+        expect(firstItem.find(AddButton).children().prop('disabled')).toBe(true);
+        expect(firstItem.find(SubButton).children().prop('disabled')).toBe(true);
+    });
+
+    it('should render an Item with key/value pass in props', () => {
+        const { component } = reduxMount(<Tags
+            objectMetadata={{ ...OBJECT_METADATA, tags: [{ key: 'key1', value: 'value1' }] }}/>);
+
+        // check if only one item row is rendered
+        const items = component.find(Item);
+        expect(items).toHaveLength(1);
+
+        // check if item row values are rendered
+        const firstItem = items.first();
+        expect(firstItem.find('input#tags-input-key').prop('value')).toBe('key1');
+        expect(firstItem.find('input#tags-input-value').prop('value')).toBe('value1');
+    });
+
+    it('should delete key/value if remove button is pressed', () => {
+        const { component } = reduxMount(<Tags objectMetadata={{
+            ...OBJECT_METADATA,
+            tags: [{ key: 'key1', value: 'value1' }, { key: 'key2', value: 'value2' }],
+        }}/>);
+
+        // check if two item rows are rendered
+        let items = component.find(Item);
+        expect(items).toHaveLength(2);
+
+        // check if first item row values are rendered
+        let firstItem = items.first();
+        expect(firstItem.find('input#tags-input-key').prop('value')).toBe('key1');
+        expect(firstItem.find('input#tags-input-value').prop('value')).toBe('value1');
+
+        // check if second item row values are rendered
+        const secondItem = items.at(1);
+        expect(secondItem.find('input#tags-input-key').prop('value')).toBe('key2');
+        expect(secondItem.find('input#tags-input-value').prop('value')).toBe('value2');
+
+        // check if first AddButton is hidden and second one is visible
+        expect(component.find(AddButton).children().at(0).prop('isVisible')).toBe(false);
```

Comment:
Also, we can add className `delbtn` to `SubButton`'s `Button` component and find `find('.delbtn')` instead of using 'children'.  It will be less mistake-prone if we happen to change  `AddButton` and `SubButton` component.
Same thing for `addbtn` to `AddButton`'s `CustomButton` component.

Reply:
done